person.rb

```diff
@@ -1,47 +1,22 @@
 #!/usr/bin/env ruby
-class Animal
-    attr_accessor :name
-    def initialize(type, number_of_legs, name = "Unknown")
-        @id = Random.rand(1..1000)
-        @name = name
-        @number_of_legs = number_of_legs
-        @type = type
-    end
-
-    def speak
-        if @type == "dog"
-            "Woof, Woof"
-        elsif @type == "spider"
-            "..."
-        end
-    end
-
-    def bring_a_stick
-        if @type =="dog"
-            "Here your stick ---------"
-        end
-    end
-
-    def make_a_web
-        if@type == "spider"
-            "www"
-        end
-    end
-
-    def id
-        @id
-    end
-
-    def type 
-        @type
-    end
-
+class Person
+  def initialize(age:, name: 'Unknown', _parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+  end
```

Comment:
- [ ] It would be better if you save the  parent_permission argument. Currently, you are not saving this argument that is why you have an underscore at the beginning of your _parent_perimision, VScose is telling you that you are not using it. 

```suggestion
  def initialize(age:, name: 'Unknown', _parent_permission: true)
    @id = Random.rand(1..1000)
    @name = name
    @age = age
    @parent_permission = parent_permission
  end
```

Reply:
Thank you for the tip